src/App.js

```diff
@@ -1,13 +1,54 @@
 import React from 'react';
 
 import './App.scss';
+import { GoodList } from './GoodList';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export class App extends React.Component {
+  state = {
+    goods: [],
+  }
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  loadAllGoods = () => {
+    getAll()
+      .then(goods => this.setState({ goods }));
+  }
 
-export default App;
+  loadFive = () => {
```

Comment:
loadFive, loadAllGoods and loadRed are three the same functions. You can create one function and pass these functions' names ({ getAll, get5First, getRedGoods }) as arguments:

  loadGoods = (func) => {
    func.then(goods => this.setState({ goods }));
  }

onClick={ pass name of a function as an argument}:

onClick={this.loadGoods(getAll)}

Reply:
Thanks)